app/events/users/user_deleted_event.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class Users::UserDeletedEvent < UserRelatedEvent
+
+  # This method smells of :reek:NilCheck
+  def self.create(user)
+    new(
+      user_id: user.id,
+      metadata: {
+        provider_id: user.provider_account&.id || user.tenant_id
```

Comment:
When it is a user of a provider, it will set that as master.id? Is that intended?

```ruby
provider = Account.providers.first!
user = provider.admins.first!
user.provider_account == Account.master
```

Reply:
It is intended, in the test I am checking that it is the master 😄 But to be honest I don't know if that is a good idea... 🤔